server/services/v1/search_indexer.go

```diff
@@ -239,3 +240,45 @@ func UnpackAndSetMD(doc map[string]interface{}, tableData *internal.TableData) {
 		tableData.UpdatedAt = internal.CreateNewTimestamp(int64(v.(float64)))
 	}
 }
+
+func FlattenObjects(data map[string]any) map[string]any {
+	resp := make(map[string]any)
+	for k, v := range data {
+		switch vType := v.(type) {
+		case map[string]any:
+			flattenObjects(k, vType, resp)
+		default:
+			resp[k] = v
+		}
+	}
+
+	return resp
+}
+
+func flattenObjects(key string, obj map[string]any, resp map[string]any) {
```

Comment:
If you add this:
```
if key != "" {
   key += schema.ObjFlattenDelimiter
}
```
you can replace below key to be `key+k` and call this function in the above directly, no need in additional loop.